util/math.go

```diff
@@ -0,0 +1,11 @@
+package util
+
+func MaxUint64(values ...uint64) (max uint64) {
```

Comment:
Nit: 
1. Could be a generic max function
2. I find it to be an antipattern to have a `util` package (becomes a code dumpster):
- suggestion: `lib/math/math.go`

Reply:
agreed, moved 
generic did not worked that easy, so I kept as MaxUint64